File: batch/src/main/java/com/isystk/sample/batch/jobs/importMst/ImportMstPostValidator.java
package com.isystk.sample.batch.jobs.importMst;

import com.isystk.sample.common.validator.AbstractValidator;
import java.util.Objects;
import org.slf4j.Logger;
import org.springframework.stereotype.Component;
import org.springframework.validation.Errors;

@Component
public class ImportMstPostValidator extends AbstractValidator<ImportMstPostDto> {

  private static final Logger log = org.slf4j.LoggerFactory.getLogger(ImportMstPostValidator.class);

  @Override
  protected void doValidate(ImportMstPostDto dto, Errors errors) {

    if (dto.getPosition() == 1 && !Objects.equals(dto.getPostTagId(), "Wyn")) {
      errors.rejectValue("postTagId", "importMstJob.invalidMstPostId");
    }
  }
}

File: batch/src/main/kotlin/com/isystk/sample/batch/jobs/importMst/ImportMstPostValidator.kt
package com.isystk.sample.batch.jobs.importMst

import com.isystk.sample.common.validator.AbstractValidator
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import org.springframework.validation.Errors

@Component
class ImportMstPostValidator : AbstractValidator<ImportMstPostDto?>() {

    companion object {
        private val log = LoggerFactory.getLogger(ImportMstPostValidator::class.java)
    }

    override fun doValidate(form: ImportMstPostDto?, errors: Errors?) {
        if (form!!.position == 1 && form.postTagId != "Wyn") {
            errors!!.rejectValue("postTagId", "importMstJob.invalidMstPostId")
        }
    }
}